bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationGCJClient.java

```diff
@@ -17,114 +17,125 @@
 import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
 import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.mockito.stubbing.Answer;
 
-@RunWith(MockitoJUnitRunner.class)
+@RunWith(JUnit4.class)
 public class TestBulkMutationGCJClient {
 
   @Rule public ExpectedException expect = ExpectedException.none();
 
-  @Mock private UnaryCallable<RowMutation, Void> callable;
+  @Rule public MockitoRule rule = MockitoJUnit.rule();
 
-  private BulkMutationGCJClient bulkMutationClient;
-  private SettableApiFuture<Void> future;
-  private RowMutation rowMutation = RowMutation.create("fake-table", "fake-key");
+  @Mock private Batcher<RowMutationEntry, Void> batcher;
 
-  @Before
-  public void setUp() {
-    BulkMutationBatcher bulkMutationBatcher = new BulkMutationBatcher(callable);
-    bulkMutationClient = new BulkMutationGCJClient(bulkMutationBatcher);
-    future = SettableApiFuture.create();
-  }
+  @InjectMocks private BulkMutationGCJClient bulkMutationClient;
+
+  private RowMutationEntry rowMutation = RowMutationEntry.create("fake-key");
 
   @Test
   public void testAdd() {
-    when(callable.futureCall(rowMutation)).thenReturn(future);
+    SettableApiFuture<Void> future = SettableApiFuture.create();
+    when(batcher.add(rowMutation)).thenReturn(future);
     ApiFuture<Void> result = bulkMutationClient.add(rowMutation);
     assertFalse(result.isDone());
     future.set(null);
     assertTrue(result.isDone());
-    verify(callable).futureCall(rowMutation);
+    verify(batcher).add(rowMutation);
   }
 
   @Test(expected = ExecutionException.class)
   public void testAddFailure() throws Exception {
-    when(callable.futureCall(rowMutation)).thenReturn(future);
+    SettableApiFuture<Void> future = SettableApiFuture.create();
+    when(batcher.add(rowMutation)).thenReturn(future);
     future.setException(new RuntimeException("can not perform mutation"));
     ApiFuture<Void> result = bulkMutationClient.add(rowMutation);
     assertTrue(result.isDone());
-    verify(callable).futureCall(rowMutation);
+    verify(batcher).add(rowMutation);
     result.get();
   }
 
   @Test
   public void testFlush() throws Exception {
+    final SettableApiFuture<Void> future = SettableApiFuture.create();
     final SettableApiFuture<Void> future2 = SettableApiFuture.create();
-    when(callable.futureCall(rowMutation)).thenReturn(future).thenReturn(future2);
+    when(batcher.add(rowMutation)).thenReturn(future).thenReturn(future2);
+    doAnswer(
+            new Answer() {
+              @Override
+              public Object answer(InvocationOnMock invocationOnMock) {
+                future.set(null);
+                future2.set(null);
+                return null;
+              }
+            })
+        .when(batcher)
+        .flush();
```

Comment:
Should we make actuals calls to verify this OR this mock test would suffice?

Reply:
I think it should suffice